drivers/hmis_csv_twenty_twenty_six/app/models/hmis_csv_twenty_twenty_six/importer/custom/custom_import_concern.rb

```diff
@@ -0,0 +1,166 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module HmisCsvTwentyTwentySix::Importer::Custom::CustomImportConcern
+  extend ActiveSupport::Concern
+  include HmisCsvTwentyTwentySix::CustomModelConfig
+
+  included do
+    def as_destination_record
+      config = self.class.custom_file_config
+      klass = self.class.reflect_on_association(:destination_record).klass
+
+      # Apply all column mappings using the generic mapper
+      mapped_attributes = {}
+      HmisCsvTwentyTwentySix::Importer::Custom::ColumnMapper.apply_mappings(
+        self,
+        mapped_attributes,
+        config['columns'],
+      )
+
+      record = klass.new(mapped_attributes)
+
+      # For augmentations, we explicitly don't update the source hash since that would cause future
+      # imports to the augmented class to appear modified
+      # We also don't set the source_id as it would cause the source data drill-down to break
+      unless self.class.augments?
+        record.source_hash = source_hash
+        # Note which record we're sending this from for error checking
+        record.source_id = id
+        # For non-augmentations, we need to set the data_source_id
+        record.data_source_id = data_source_id
+      end
```

Comment:
minor, maybe a little easer to read with early-exit (not tested)
```suggestion
      # For augmentations, we explicitly don't update the source hash since that would cause future
      # imports to the augmented class to appear modified
      # We also don't set the source_id as it would cause the source data drill-down to break
      return record if self.class.augments?

      # For new_table, set the source hash, source ID, and data source ID.
      record.source_hash = source_hash
      # Note which record we're sending this from for error checking
      record.source_id = id
      # For non-augmentations, we need to set the data_source_id
      record.data_source_id = data_source_id

      record
```

Reply:
Looks good to me.